src/game_initialization/multiplayer.cpp

```diff
@@ -572,18 +581,30 @@ bool mp_manager::enter_lobby_mode()
 	return true;
 }
 
-void mp_manager::enter_create_mode()
+void mp_manager::enter_create_mode(bool preset, const std::string& preset_scenario)
 {
 	DBG_MP << "entering create mode";
 
-	if(gui2::dialogs::mp_create_game::execute(state, connection == nullptr)) {
-		enter_staging_mode();
+	config presets;
+	if(preset) {
+		for(const config& game : game_config_manager::get()->game_config().mandatory_child("mp_queue").child_range("game")) {
+			if(game["mp_scenario"] == preset_scenario) {
+				presets = game;
+			}
+		}
+	}
+
+	if(preset) {
```

Comment:
This can be merged with the if above, no?

Reply:
Updated.